mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -6,6 +6,8 @@
 import urllib.parse
 from mimetypes import guess_type
 
+from botocore.exceptions import ClientError
```

Comment:
Do we need this as a top-level import? Can this be a localized import within the calling function to eliminate the need to include this as a dependency?

Reply:
Trying that now.